server/mailrepository/mailrepository-api/src/test/java/org/apache/james/mailrepository/MailRepositoryTest.java

```diff
@@ -0,0 +1,125 @@
+package org.apache.james.mailrepository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.StringJoiner;
+
+import javax.mail.MessagingException;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailrepository.api.MailRepository;
+import org.apache.james.mime4j.io.InputStreams;
+import org.apache.james.server.core.MailImpl;
+import org.apache.mailet.Mail;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public interface MailRepositoryTest {
+
+    String TEST_ATTRIBUTE = "testAttribute";
+
+    default  Mail createMail(String name) throws MessagingException {
+        InputStream mailContent = generateMailContent();
+        List<MailAddress> recipients = ImmutableList
+            .of(new MailAddress("rec1@domain.com"),
+                new MailAddress("rec2@domain.com"));
+        MailAddress sender = new MailAddress("sender@domain.com");
+        Mail mail = new MailImpl(name, sender, recipients, mailContent);
+        mail.setAttribute(TEST_ATTRIBUTE, "testValue");
+        return mail;
+    }
+
+    default InputStream generateMailContent() {
+        String headers = new StringJoiner("\r\n")
+            .add("Subject: test")
+            .add("Content-Type: text/plain")
+            .toString();
+        String headerBodySeparator = "\r\n\r\n";
+        String body = "original body" + RandomStringUtils.randomAlphabetic(10);
+        String end = "\r\n.\r\n";
+        return InputStreams.create(headers + headerBodySeparator + body + end, StandardCharsets.UTF_8);
+    }
+
+    default void checkMailEquality(Mail actual, Mail expected) {
+        assertAll(
+            () -> assertThat(actual.getMessage().getContent()).isEqualTo(expected.getMessage().getContent()),
+            () -> assertThat(actual.getMessageSize()).isEqualTo(expected.getMessageSize()),
+            () -> assertThat(actual.getName()).isEqualTo(expected.getName()),
+            () -> assertThat(actual.getState()).isEqualTo(expected.getState()),
+            () -> assertThat(actual.getAttribute(TEST_ATTRIBUTE)).isEqualTo(expected.getAttribute(TEST_ATTRIBUTE))
```

Comment:
You miss many fields: lastUpdated, errormessage,  remoteAddr, remoteHost, etc...

Reply:
No I don't think so. the commit is "port existing tests" and the existing test does exactly these assertions. I'll do more when I'll be covering MailRepository more seriously.